test/e2e/bundle_e2e_test.go

```diff
@@ -70,9 +73,19 @@ var _ = Describe("Installing bundles with new object types", func() {
 
 			// Wait for the Subscription to succeed
 			Eventually(func() error {
-				_, err = fetchSubscription(operatorClient, testNamespace, subName, subscriptionStateAtLatestChecker)
+				sub, err := fetchSubscription(operatorClient, testNamespace, subName, subscriptionStateAtLatestChecker)
+				if sub != nil && sub.Status.InstallPlanRef != nil {
+					installPlanRef = sub.Status.InstallPlanRef.Name
+				}
+				return err
+			}).Should(BeNil())
+
+			// Wait for the installplan to complete (5 minute timeout)
+			Eventually(func() error {
+				_, err := fetchInstallPlan(GinkgoT(), operatorClient, installPlanRef, buildInstallPlanPhaseCheckFunc(v1alpha1.InstallPlanPhaseComplete))
```

Comment:
Can a subscription be at latest if the InstallPlan isn't complete?

Reply:
I assumed so but let me double check - that's basically my concern with this fix

Reply:
From what I can see the, the answer is yes. The `SubscriptionStateAtLatest` is concerned only with whether an update exists for that given CSV rather than the actual status of its InstallPlan. 

see https://github.com/operator-framework/operator-lifecycle-manager/blob/e72d31aee442945cd9757f47bd075ec15ec0ea1c/pkg/controller/operators/catalog/operator.go#L939